custom_components/rpi_gpio/cover.py

```diff
@@ -104,10 +104,18 @@ def __init__(self, hub, name, relay_port, relay_time, relay_active_low, relay_bi
         self._relay_drive = relay_drive
         self._state_port = state_port
         self._state_bias = state_bias
-        self._start_active_low = state_active_low
+        self._state_active_low = state_active_low
         self._attr_is_closed = False != state_active_low
-        hub.add_cover(self, relay_port, relay_active_low, relay_bias, relay_drive,
-                      state_port, state_bias, state_active_low)
+
+    async def async_added_to_hass(self) -> None:
+        await super().async_added_to_hass()
+        self._hub.add_cover(self, self._relay_port, self._relay_active_low, self._relay_bias, 
+                            self._relay_drive, self._state_port, self._state_bias, self._state_active_low)
+        self.async_write_ha_state()
+
+    # dirty hack to enable reuse of switch
+    def is_on(self):
+        return False
```

Comment:
Do you understand why this is needed? I think upstream repo used this for something else which is not used here.
The cover entity doesn't access this attribute

Reply:
I have to admit I didn't check that. @jdeneef might shade some light on this. I can try to find time tomorrow to debug it a bit.

Reply:
@thecode
```add_cover() -> add_switch() -> entity.is_on```
So this is why we need is_on implemented.

Reply:
This is really a dirty hack which makes the code unreadable and doesn't save any lines of code. It would be simpler to just add the relevant code ([Lines 141-148](https://github.com/thecode/ha-rpi_gpio/blob/5713ae8deb46dbfc584b1d90696ad192a8518d63/custom_components/rpi_gpio/hub.py#L141C1-L148C10)):
```python
        self._entities[port] = entity
        self._config[port] = gpiod.LineSettings(
            direction = Direction.OUTPUT,
            bias = BIAS[bias],
            drive = DRIVE[drive_mode],
            active_low = active_low,
            output_value = Value.INACTIVE
        )
```
Instead of calling `add_switch` at [Line 185](https://github.com/thecode/ha-rpi_gpio/blob/5713ae8deb46dbfc584b1d90696ad192a8518d63/custom_components/rpi_gpio/hub.py#L185)

Sensor could also be removed, so your end `add_cover` will look like this:
```python
def add_cover(self, entity, relay_port, relay_active_low, relay_bias, relay_drive, 
                  state_port, state_bias, state_active_low) -> None:
        _LOGGER.debug(f"in add_cover {relay_port} {state_port}")
        self._entities[relay_port] = entity
        self._config[relay_port] = gpiod.LineSettings(
            direction = Direction.OUTPUT,
            bias = BIAS[relay_bias],
            drive = DRIVE[relay_drive],
            active_low = relay_active_low,
            output_value = Value.INACTIVE
        )       
        self.add_sensor(entity, state_port, state_active_low, state_bias, 50)
        self.update_lines()
```

Using the `is_on` is a way to make it break in the future since HA team constantly add validations for entity properties.

Reply:
Agree that we can remove the hack, I went a little bit different way which I think gives us the benefits of both approaches. Let me know what you think.